### ISSUE TITLE
DNN

### ISSUE THREAD

**Issue by iankuoli** [11thDeepLearningGroup2015/MLDS_HW1#10]
Implement a DNN
- [ ] Randomized initialization according to [X. Bengio et al., 2010].
- [ ] Using tanh as the activation function of hidden layers.
- [ ] Using softmax as the activation function of the last layer.
- [ ] Test my model by learning the MNIST dataset.
- [ ] Test my model by learning the TIMIT dataset (the subject of this project).
- [ ] Explore a better optimization method.

**Comment by iankuoli** [comment]
test data error rate: 53%

**Comment by wush978** [comment]
So the FER is 0.47?

**Comment by iankuoli** [comment]
i think the accuracy rate should be 0.47 (error = 0.52XX)
my model uses 4 hidden layer each of which has 500 neurons.
learning rate = 0.01

**Comment by iankuoli** [comment]
Now i do not use instance-wise normalization.
i will use it in the next run.

**Comment by wush978** [comment]
請教一下, Kaggle上的是TIMIT dataset嗎？我還沒有完全狀況內。MNIST dataset在哪裡阿？

**Comment by iankuoli** [comment]
MNIST 是 http://www.deeplearning.net/tutorial/logreg.html 這裡的data，我拿來測試程式正確與否。
TIMIT dataset是從Kaggle上下載來的，用你sample1的切法測

**Comment by wush978** [comment]
Got it ! Thanks

**Comment by ccwang002** [comment]
你的 batch size 跟 epoch（iteration）是多少？因為用的 neuron 已經比老師多了，感覺 error rate 應該有辦法更低。你看能不能畫一張 error rate 隨著 epoch 的變化，training 跟 testing 各一張，看一下掉下來的趨勢。

**Comment by iankuoli** [comment]
ok. 昨天最好 error rate到0.47左右

**Comment by iankuoli** [comment]
另外想請教一下，我用instance-wise normalization 結果差很多，你們有試過用instance-wise normalization跑的結果嗎？我怕我寫錯了。

**Comment by wush978** [comment]
我都是用instance-wise normalization

**Comment by ccwang002** [comment]
我是用 feature-wise normalization，也許可以比較一下？

**Comment by iankuoli** [comment]
我用instance-wise normalization error rate 0.8x
我的方法是把所有資料存在矩陣裡，每個instance的feature vector是一個row
對矩陣做column normalized

**Comment by iankuoli** [comment]
epoch 1, minibatch 44992/44992, validation error 57.574680 %
epoch 2, minibatch 44992/44992, validation error 54.382112 %
epoch 3, minibatch 44992/44992, validation error 52.972084 %
epoch 4, minibatch 44992/44992, validation error 51.845661 %
epoch 5, minibatch 44992/44992, validation error 51.319346 %
epoch 6, minibatch 44992/44992, validation error 50.978841 %
epoch 7, minibatch 44992/44992, validation error 51.036629 %
epoch 8, minibatch 44992/44992, validation error 50.403627 %
epoch 9, minibatch 44992/44992, validation error 50.778805 %
epoch 10, minibatch 44992/44992, validation error 50.676565 %
epoch 11, minibatch 44992/44992, validation error 49.612376 %
epoch 12, minibatch 44992/44992, validation error 49.942212 %
epoch 13, minibatch 44992/44992, validation error 48.890469 %
epoch 14, minibatch 44992/44992, validation error 49.594595 %
epoch 15, minibatch 44992/44992, validation error 48.585526 %
epoch 16, minibatch 44992/44992, validation error 48.795341 %
epoch 17, minibatch 44992/44992, validation error 48.421942 %
epoch 18, minibatch 44992/44992, validation error 47.924964 %
epoch 19, minibatch 44992/44992, validation error 48.076991 %
epoch 20, minibatch 44992/44992, validation error 48.425498 %
epoch 21, minibatch 44992/44992, validation error 47.195057 %
epoch 22, minibatch 44992/44992, validation error 46.912340 %
epoch 23, minibatch 44992/44992, validation error 47.829836 %
epoch 24, minibatch 44992/44992, validation error 47.014580 %
epoch 25, minibatch 44992/44992, validation error 47.011913 %
epoch 26, minibatch 44992/44992, validation error 48.092994 %
epoch 27, minibatch 44992/44992, validation error 47.147048 %
epoch 28, minibatch 44992/44992, validation error 47.125711 %
epoch 29, minibatch 44992/44992, validation error 47.334637 %
epoch 30, minibatch 44992/44992, validation error 46.779872 %

**Comment by ccwang002** [comment]
感覺可以後調高 learning rate，epoch 數再高一點 error rate 應該掉得下來。

minibatch 是每個 epoch 全部 sample 都用嗎？這邊看不懂

**Comment by iankuoli** [comment]
恩 全部 sample 都用

**Comment by ccwang002** [comment]
可以試把 minibatch 調小一點，150、300 之類

**Comment by iankuoli** [comment]
最新成績： error 37.916074 %
8 layers, learning rate: 0.0005, without L1 or L2 regularzation
